addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,719 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import base64
+import io
+import re
+import time
+import zipfile
+import urllib.parse
+import uuid
+import requests
+
+from dateutil.relativedelta import relativedelta
+from datetime import datetime
+from requests import RequestException
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+SINVOICE_API_URL = 'https://api-vinvoice.viettel.vn/services/einvoiceapplication/api/'
+SINVOICE_TIMEOUT = 60  # They recommend between 60 and 90 seconds, but 60s is already quite long.
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    # EDI values
+    l10n_vn_edi_state = fields.Selection(
+        string='Sinvoice Status',
+        selection=[
+            ('ready_to_send', 'Ready to send'),
+            ('sent', 'Sent'),
+            # Set when we write on the payment status
+            ('payment_state_to_update', 'Payment status to update'),
+            ('cancelled', 'Cancelled'),
+            ('adjusted', 'Adjusted'),
+            ('replaced', 'Replaced'),
+        ],
+        copy=False,
+        compute='_compute_l10n_vn_edi_state',
+        store=True,
+        readonly=False,
+    )
+    # This id is important when sending by batches in order to recognize individual invoices.
+    l10n_vn_edi_transaction_id = fields.Char(
+        string='SInvoice Transaction ID',
+        help='Technical field to store the transaction ID if needed',
+        export_string_translation=False,
+        copy=False,
+    )
+    l10n_vn_edi_invoice_symbol = fields.Many2one(
+        string='Invoice Symbol',
+        comodel_name='sinvoice.symbol',
+        compute='_compute_l10n_vn_edi_invoice_symbol',
+        readonly=False,
+        store=True,
+    )
+    l10n_vn_edi_invoice_number = fields.Char(
+        string='Invoice Number',
+        help='Invoice Number as appearing on SInvoice.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_reservation_code = fields.Char(
+        string='Secret Code',
+        help='Secret code that can be used by a customer to lookup an invoice on SInvoice.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_issue_date = fields.Datetime(
+        string='Issue Date',
+        help='Date of issue of the invoice on the e-invoicing system.',
+        copy=False,
+        readonly=True,
+    )
+    l10n_vn_edi_sinvoice_file_id = fields.Many2one(
+        comodel_name='ir.attachment',
+        compute=lambda self: self._compute_linked_attachment_id('l10n_vn_edi_sinvoice_file_id', 'l10n_vn_edi_sinvoice_file'),
+        depends=['l10n_vn_edi_sinvoice_file'],
+        export_string_translation=False,
+    )
+    l10n_vn_edi_sinvoice_file = fields.Binary(
+        string='SInvoice json File',
+        copy=False,
+        export_string_translation=False,
+    )
```

Comment:
That's how it was done for most EDI's since the new send & print, so I did it the same way. I'm not exactly sure about the need for that since I never found any explanations on why it was done this way. I thought it had perf/... benefits

> I don't know why you put export_string_translation on all fields... Binary/Many2one are not translatable

What do you mean, of course they are. `translate` != `export_string_translation`. Here what I want is to avoid exporting these strings in the POT file since it would be a waste of time to translate the labels as they won't be shown in the view.